### PR TITLE
Add monitoring resources to scalardb-graphql chart

### PR DESCRIPTION
**Pull request by ymorimo**
This PR adds Service, ServiceMonitor, and PrometheusRule resources to `scalardb-graphql` chart as `scalardb` and `scalardl` charts do.

The latest build of the docker image from the `scalardb-graphql` `main` branch is needed.

I confirmed on a minikube cluster:

* Build a docker image of scalardb-graphql

    ```console
    cd /path/to/scalardb-graphql
    ./gradlew docker
    minikube image load ghcr.io/scalar-labs/scalardb-graphql:4.0.0-SNAPSHOT
    minikube image tag ghcr.io/scalar-labs/scalardb-graphql:4.0.0-SNAPSHOT local/test/scalardb-graphql:4.0.0-SNAPSHOT
    ```

* Start a database in the `minikube` network and load a schema

    ```console
    docker run -d --name cassandra-scalardb --network minikube -p 9042:9042 cassandra:3.11
    java -jar scalardb-schema-loader-3.5.0.jar ...
    ```

* Create a custom values file

    ```
    cat <<EOF > test-custom-values.yaml
    image:
      repository: local/test/scalardb-graphql
      tag: 4.0.0-SNAPSHOT
    
    scalarDbGraphQlConfiguration:
      contactPoints: cassandra-scalardb
      contactPort: 9042
      username: cassandra
      password: cassandra
      storage: cassandra
      logLevel: INFO
      path: /graphql
      namespaces: emoney
      graphiql: "true"
    
    service:
      type: LoadBalancer

    serviceMonitor:
      enabled: true
    prometheusRule:
      enabled: true
    ingress:
      enabled: true
      className: nginx
    EOF
    ```
    
* Deploy using Helm

    ```console
    helm install scalardb-graphql ./charts/scalardb-graphql/ -f /path/to/test-custom-values.yaml
    ```

Then, confirmed that monitoring is working by following the guide: https://github.com/scalar-labs/helm-charts/blob/main/docs/getting-started-monitoring.md

---

The Grafana Dashboard configuration is going to be added in another PR since it will be large.

### CODE REVIEW

File: charts/scalardb-graphql/templates/scalardb-graphql/service.yaml [hunk 1]
```diff
@@ -16,3 +16,19 @@ spec:
...
+kind: Service
+metadata:
+  name: {{ include "scalardb-graphql.fullname" . }}-metrics
+  labels:
+    {{- include "scalardb-graphql.labels" . | nindent 4 }}
+spec:
+  ports:
+  - name: scalardb-graphql-prometheus
+    port: 9080
+    targetPort: 9080
```
Comment (ymorimo) [inline charts/scalardb-graphql/templates/scalardb-graphql/service.yaml:30]: The Prometheus port `9080` differs from `8080` on other charts because `8080` is used as the default port of the GraphQL HTTP server. I feel it was a design mistake. I wonder if we should change the GraphQL HTTP port to another number such as `8000`.

Comment (kota2and3kan) [reply charts/scalardb-graphql/templates/scalardb-graphql/service.yaml:30]: This is a minor comment for your reference. This comment does not prevent for merging the PR.

I think the GraphQL HTTP port is used for GraphiQL that is accessed from browsers as follows.

* GraphiQL URL
  ```
  http://localhost:8080/graphql
  ```

Also, the default port of Web UI of Scalar Manager is `8000` now.
The Scalar Manager Web UI is accessed from browsers as follows.
* Scalar Manager Web UI URL
  ```
  http://localhost:8000/
  ```

So, if we change the default port of GraphQL HTTP to `8000`, I think it causes another conflict of port between the GraphiQL and the Scalar Manager Web UI.
Basically, I think both will be exposed to the outside of Kubernetes Cluster to access them from browsers.

I think we need to consider the above thing when we decide the default port.

**Review by kota2and3kan — APPROVED** [review]
LGTM! Thank you!

I left a minor comment that is not related to this PR directly.
Please take a look!

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!
